core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/ProblemStatistics.java

```diff
@@ -0,0 +1,12 @@
+package ai.timefold.solver.core.api.solver;
+
+/**
+ * The statistics of a given problem submitted to a {@link Solver}.
+ *
+ * @param entityCount The number of genuine entities defined by the problem.
+ * @param variableCount The number of genuine variables defined by the problem.
+ * @param maximumValueRangeSize The number of possible assignments for the genuine variable with the largest range.
+ * @param problemScale An approximate log of the solution space.
```

Comment:
I think this last line deserves a bit more description. What do you mean by `log`, a logarithm? A natural logarithm? What is a solution space? Maybe it'd be best if we just say what the formula is.

Reply:
Pending @ge0ffrey decision to change the meaning of `problemScale`  from `entityCount * valueCount` (that the current definition) to `log(product(value_range_size_for_entity)) = log(value_range_size_for_entity_1) +  log(value_range_size_for_entity_2) + ... ~= log(value_range_size) * entityCount ` (this is for basic variables, list/chained variables have a different counting system to discard invalid assignments)